### ISSUE TITLE
Error when analyzing packages without go.mod

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#1027]
if the package under analysis hasn't a `go.mod` file, `revive` stops with the error 
```
linting - retrieving failures channel: "<package full path>" doesn't seem to be part of a Go module
```

**To Reproduce**
1. I updated `revive` `go install github.com/mgechev/revive@latest`
1. Check out [vim-go](https://github.com/fatih/vim-go)
1. Remove or rename the file `vim-go/autoload/go/test-fixtures/lint/src/errcheck/go.mod`
3. From the `go-vimI` directory, run `revive`:

```shell
revive ./...
```

**Expected behavior**
`revive` should analyze the go sources without reporting the above error

**Comment by chavacava** [comment]
@bhcleek could you please confirm that the PR #1028 fix the problem?
